tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/interceptor/FaultToleranceInterceptorTest.java

```diff
@@ -84,27 +67,34 @@ public static WebArchive deploy() {
      * @throws InterruptedException
      * @throws ExecutionException
      */
-    @Test
+    @Test(priority = 1)
     public void testAsync() throws InterruptedException, ExecutionException {
         Future<String> result = testInterceptor.asyncGetString();
         assertEquals(result.get(), "OK");
-        orderKeeper.poll();
         String [] expectedOrder = {"EarlyOrderFtInterceptor","LateOrderFtInterceptor","asyncGetString"};
-        assertEquals(orderKeeper.toArray(), expectedOrder);
+        assertEquals(orderFactory.getOrderQueue().toArray(), expectedOrder);
     }
 
-    @Test
+    @Test(priority = 2)
     public void testRetryInterceptors() {
         try {
-            testInterceptor.serviceA();
+            testInterceptor.serviceRetryA();
             fail("Exception not thrown");
         }
         catch (Exception e) {
 
         } // Expected
 
-        assertEquals(methodCounter.get(), 6, "methodCounter"); // Method called six times (1 + 5 retries)
-        assertEquals(earlyInterceptorCounter.get(), 2, "earlyInterceptorCounter");
-        assertEquals(lateInterceptorCounter.get(), 7, "lateInterceptorCounter");
+        String [] expectedOrder = {"EarlyOrderFtInterceptor","LateOrderFtInterceptor","serviceRetryA",
+            "LateOrderFtInterceptor","serviceRetryA"}; //executes 1 more time the later interceptor and the bean method
+        // annotated with maxRetries = 1
+        assertEquals(orderFactory.getOrderQueue().toArray(), expectedOrder);
+    }
+
+    @AfterTest
```

Comment:
I think you want `@AfterMethod` here rather than `@AfterTest`

A "test" in TestNG is a larger grouping which can include lots of test classes (I find this super-confusing because I'm used to thinking about each method as a "test")